### DISCUSSION TITLE
Development still on going?

### DISCUSSION THREAD

**Discussion by REMC19** [Aries85/LightZone#341] [General]
Masahiro
It's been a while since there appear to have been any updates to LightZone on GitHub.
Are you still developing it - i.e. is this apparent hiatus due to other commitments?
Just wondering.

**Comment by ktgw0316** [comment]
@REMC19 
I understand your concern. I've been very busy at work, and I haven't been able to make progress on LZ these days. I hope that the situation will be better in August.

**Comment by doobes** [comment]
Was wondering the same thing. 

Good luck. 

If we can help with anything, please let us know. 

Chris

**Comment by doobes** [comment]
Just chiming in for a show of support for your efforts.

**Comment by REMC19** [comment]
Hi Masahiro
I know I've asked this before but it's been a while again since there appear to have been any updates to LightZone on GitHub, while you have been active on other things.
Are you still developing it - (i.e. intending to get back to it at some point) or have you decided it time to call it a day as far the development of LighZone is concerned?
Either way, thanks for all the effort you have put in to keeping it going.

**Reply by ktgw0316** [reply]
As you know, I've been busy on another project these days.

The current beta version of LightZone for Windows works on my PC when Gradle is used, but after installation, it fails to load DLLs and cannot open images. I think we need the help of other developers who are skilled in Windows.

I'll write about Linux and macOS in this weekend. Does LZ work fine on these OS on your machine? I need feedback to understand the situation, since sometimes it doesn't work on your side even if it works fine on my development environment.

**Reply by ktgw0316** [reply]
- LZ for macOS can open and edit RAW files on m1 MacBook Air, but "check for update" and "LightZone Help" is not working.
- LZ for Linux runs fine, at least on my environment (Ubuntu 24.04 LTS on WSL2, and Arch Linux).
- Debian doesn't support Gradle Kotlin scripts; it only supports older Gradle scripts written in the Groovy language. This makes it difficult to migrate from Ant to Gradle. (The migration has already done on Windows and macOS.)

**Reply by sfink16** [reply]
> * LZ for Linux runs fine, at least on my environment (Ubuntu 24.04 LTS

Yes, it works my Ubuntu 24.04 as well, but not on my Windows 10.  I don't see an error message but I'm not very knowledgeable as to why I don't see a message because I rarely get on that desktop.

**Comment by doobes** [comment]
This is going to sound very outer space, but is there any sense in thinking about converting the Java code to C#?

Doing so might open up resources for others to aid in the development, taking some of the load off of Masahiro.

My real fear is that in its present state, LightZone is only barely alive and if we want to see it live on, much less thrive, then desperate choices might be necessary.

Chris

**Reply by ktgw0316** [reply]
@doobes I understand your concern. Having only one developer is indeed not an ideal situation for an OSS project. I've been working to migrate from Ant to Gradle to make builds easier for anyone to perform. We should also change the programming language if doing so would further improve the situation.

If you want to convert the code to a language that is more modern than Java, Kotlin would be the best choice. In that case, you can replace part of the code at a time, but you need to rewrite the entire code at once if you choose a non-JVM language, such as C#. I think making gradual improvements is more realistic than rebuilding everything at once.

**Reply by doobes** [reply]
Thank you for the reply.

Please understand, I'm not wanting to be antagonistic, not in any way, shape or form.  I think what you have done keeping this project alive is simply remarkable. I also understand that you have a life, and that will always be more important in the global scheme of things than LightZone.

In reality, I'm trying to start a dialog. I am not a developer and don't play one on TV (sorry, old joke). What I am is a retired electro-mechanical engineer who could possibly assist in managing a "proposed" transition.

Let's keep this dialog going.  I've started a similar one over at pixls.us in the LightZone thread.  Please feel free to interject there.

Be well

chris
